stdlib/public/core/StringLegacy.swift

```diff
@@ -34,13 +34,19 @@ extension String {
       return
     }
 
-    // TODO(String performance): We can directly call appendInPlace
```

Comment:
It appears the original author intended to call the internal method `appendInPlace`, but this PR doesn't do that. Can you explain a bit about how your approach compares to `appendInPlace`?